### PR TITLE
Refactor controller logic

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Refactoring submission controller
- Added service to generate submission
- Set string ids to be guid in openapi spec

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Running the app locally + unit tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/error/ErrorResponse.java [hunk 1]
```diff
@@ -1,20 +1,20 @@
 package ca.bc.gov.open.jag.efilingapi.error;
 
 public enum ErrorResponse {
-    INVALIDROLE("INVROLE", "User does not have a valid role for this request."),
-    ACCOUNTEXCEPTION("MLTACCNT", "Client has multiple CSO profiles"),
-    GETPROFILESEXCEPTION("CSOERROR", "Calling CSO accountFacade.getProfiles caused an exception.");
+    INVALIDROLE( "User does not have a valid role for this request."),
+    ACCOUNTEXCEPTION("Client has multiple CSO profiles"),
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/error/ErrorResponse.java:5]: non-blocking: missing period if you use the other errors, or the others have a period. vice versa.

Comment (akroon3r) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/error/ErrorResponse.java:5]: line 7 as well

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java [hunk 2]
```diff
@@ -1,18 +1,48 @@
...
+
+    @Bean
+    public SubmissionMapper submissionMapper() {
+        return new SubmissionMapperImpl();
+    }
+
     @Bean
-    public SubmissionService submissionService() {
-        return new SubmissionServiceImpl();
+    // Supressing Method submissionService has 5 arguments (exceeds 4 allowed). This is required for this service, further refactoring is needed to fix this.
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java:35]: :shipit:

Comment (akroon3r) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java:35]: was wondering if we could build a hashmap of the params before passing them in, but its not a huge deal.

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,83 @@
...
 
     @Override
-    @Cacheable(cacheNames = "submission", key = "#key", cacheManager = "submissionCacheManager", unless="#result == null")
-    public Optional<Submission> getByKey(UUID key) {
-        return Optional.empty();
+    public Submission generateFromRequest(GenerateUrlRequest generateUrlRequest) throws NestedEjbException_Exception {
+
+        logger.debug("Attempting to get user cso account information");
+        AccountDetails accountDetails = efilingAccountService.getAccountDetails(formatUserGuid(generateUrlRequest.getUserId()));
+        logger.info("Successfully got cso account information");
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java:52]: should these logs both be the same level?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/error/ErrorResponse.java (hunk 1 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/error/ErrorResponse.java]: fixed thanks!

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java (hunk 2 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java:35]: I need to remove this

Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfig.java]: fixed

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GetSubmissionTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                .create();
+
+        Mockito.when(submissionStoreMock.getByKey(CASE_3)).thenReturn(Optional.of(submissionWithoutCsoAccount));
+
+        sut = new SubmissionApiDelegateImpl(submissionServiceMock, generateUrlResponseMapperMock, navigationProperties, submissionStoreMock);
+
+    }
+
+    @Test
+    @DisplayName("With null redis storage response return NotFound")
```
Comment (akroon3r) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GetSubmissionTest.java:92]: non-blocking: display names with http status codes in the previous file are a bit more descriptive than this one.

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java (hunk 3 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java:52]: the first one is more to tell you where you are... but we can discuss that, usually debug are not enable so we don't see them...

**Pull request merged**